### PR TITLE
fix: deduplicate --model/--agent flags in CLI builder (#404)

### LINKED ISSUE #404: bug: additionalArgs can duplicate --model and --agent flags
## Bug

buildLaunchCommandForConfig() in copilot-cli-builder.ts merges per-agent additionalArgs and global editless.cli.additionalArgs into the command, but doesn't strip flags that are already being set by the builder:

- --model is set from config.model (line 110), but could also appear in additionalArgs
- --agent is derived from config.id/universe (lines 90-97), but could appear in additionalArgs
- Global and per-agent args could also duplicate each other

### Expected behavior
The builder should deduplicate known flags (--model, --agent) - per-agent values should win over global, and explicit config fields should win over additionalArgs.

### Current behavior
All args are naively concatenated, so the CLI gets duplicate flags (e.g. --model gpt-4 --model claude-sonnet). Behavior depends on how ghcp handles duplicate flags (likely last-wins, but undefined).

### Location
src/copilot-cli-builder.ts:103-111

### PR DESCRIPTION
**Pull request by cirvine-MSFT**
Closes #404
Working as Morty (Extension Dev)

## Changes
- Strip --model from additionalArgs when config.model is set
- Strip --agent from additionalArgs when agentFlag is derived from config
- Add tests for dedup edge cases

## Testing
All existing + new tests pass (830 total)

### CODE REVIEW

File: src/copilot-cli-builder.ts [hunk 1]
```diff
@@ -100,13 +100,35 @@ export function buildLaunchCommandForConfig(config: { id: string; universe: stri
...
+        // Skip dangling value (next arg if it's not a flag)
+        if (i + 1 < allExtra.length && !allExtra[i + 1].startsWith('--')) {
+          i++;
+        }
+      } else {
+        filtered.push(allExtra[i]);
+      }
+    }
+    allExtra = filtered;
+  }
```
Comment (cirvine-MSFT) [inline src/copilot-cli-builder.ts]: So we can specify additonal args at the global level or the agent level, so do we need to dedup the other flags, or just these cause it will cause major issues that may not be apparent to the user? e.g. I specify --yolo in both the global additonal args and at the agent config level?
